src/client/lixa_tx.c

```diff
@@ -53,6 +53,23 @@
 #endif /* LIXA_TRACE_MODULE */
 #define LIXA_TRACE_MODULE   LIXA_TRACE_MOD_CLIENT_TX
 
+static const char *tx_open_profile = NULL;
+
+int lixa_tx_set_profile(const char *profile) {
+    if (tx_open_profile) {
+        free((char*)tx_open_profile);
+    }
+    if (profile) {
+	tx_open_profile = strdup(profile);
```

Comment:
For this change I think it is sufficient to have a single process-wide profile, exactly as would be set by the environment variable. Effectively this replaces using setenv("LIXA_PROFILE",...). If there was a per-thread setting (which might be useful) a process-wide override from the environment variable would probably not make sense. However, I have changed the function to make it thread-safe (as "if (x) free(x)" is not) and add a lixa_tx_get_profile() which provides a safe copy to use.

Reply:
I don't agree on this point.
Once tx_set_profile() is available, a programmer using LIXA is legitimate to:

1. unset LIXA_PROFILE environment variable
2. use tx_set_profile() at thread level to have different threads using different resource manager configurations

I prefer not to speculate if this is "good" or "bad", it's just possible and a rock solid library must support it.
Using a single common variable, protected by a mutex, does not provide any benefit in avoiding the following race condition:

Thread1:
    tx_set_profile(PROF1);

Thread2:
   tx_set_profile(PROF2);
   tx_open();

Thread1:
   tx_open();

Thread1 requires PROF1, but due to the pre-emption happened between tx_set_profile(PROF1) and tx_open(), it's forced to use PROF2. This possible behavior is wrong and very difficult to debug as every race condition that happens only in some cases.
The LIXA library has its own architecture to deal with it throw the "client_status_s" as explained above; sorry but no shortpath is acceptable.

Reply:
For our use case, a startup setting of the profile (before any tx_open is called) is sufficient and all threads will use the same profile. This is equivalent to calling setenv("LIXA_PROFILE",...) before first calling tx_open.
Using setenv() is also not thread-safe if multiple threads require different profiles, though one could wrap the setenv and tx_open with a mutex *if* you can be sure that tx_open will not retain a pointer to the environment string.

Reply:
I understand your use case is fine for you, but I have to think about all the foreseeable use cases other users may implement once the new function will be available in the master repository.
If you are not interested in aligning with the architecture of the LIXA library, it will be my work to implement the feature in the way I think correct, but it will take some time: I don't commit on time, only on quality.

Reply:
Having per-thread profiles just adds complication, such as moving the digest from global_ccc to per-thread cs, not to mention writing multi-thread tests, when as far as I know there is no demand for this.
I have a further branch in preparation which allows a client to add individual RMs to the global profile. I think this is a more versatile solution, and perhaps once it is fully implemented it could be expanded to per-thread configuration.